From 3cf2baf6cfcc5dc2cf67bddc82beac12c57d9e0b Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Thu, 15 Nov 2012 11:10:26 +0200
Subject: [PATCH 0355/1825] Fix SPI functionality for KW40

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit d5a6d3546eb369f9629d78859fef83fb1467f93a

	1. Enabled compilation of mv_mtd folder by moving it from
           drivers/mtd/nand/Makefile to driver/mtd/Makefile
	2. Enabled compilation of flashmap.c by modifying the
	   the condtion from AXP to SPI

Change-Id: I63edbb29eccb68fb31b17641e07e700c69c7dd50
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Signed-off-by: Kosta Zertsekel <konszert@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c |    4 ++--
 drivers/mtd/Makefile                               |    2 ++
 drivers/mtd/maps/Makefile                          |    6 +++---
 drivers/mtd/nand/Makefile                          |    1 -
 4 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c
index 3458e99..534e93d 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c
@@ -131,7 +131,7 @@ static struct mtd_info *sflash_probe(struct map_info *map)
 	memset(mtd, 0, sizeof(*mtd));
 	memset(sflash, 0, sizeof(*sflash));
 	    
-	DB(printk("\nINFO: %s - Base address %08x",__FUNCTION__, sflash->baseAddr));
+	DB(printk("\nINFO: %s - Base address %08x\n",__FUNCTION__, map->phys));
 #ifdef CONFIG_ARCH_FEROCEON_ORION	
 	/* First check that SPI bus mode is configured to connect to an external SFlash */
     if (mvCtrlSpiBusModeDetect() != MV_SPI_CONN_TO_EXT_FLASH)
@@ -144,7 +144,7 @@ static struct mtd_info *sflash_probe(struct map_info *map)
 #endif
 	/* Try to detect the flash and initialize it over SPI */	
 	sflash->baseAddr         = map->phys;
-    sflash->index            = MV_INVALID_DEVICE_NUMBER; /* will be detected in init */	
+	sflash->index            = MV_INVALID_DEVICE_NUMBER; /* will be detected in init */	
 	sflash_disable_irqs(flags, sflash_in_irq);	
 	if (mvSFlashInit(sflash) != MV_OK)
 	{
diff --git a/drivers/mtd/Makefile b/drivers/mtd/Makefile
index f901354..0b2d280 100644
--- a/drivers/mtd/Makefile
+++ b/drivers/mtd/Makefile
@@ -33,3 +33,5 @@ inftl-objs		:= inftlcore.o inftlmount.o
 obj-y		+= chips/ lpddr/ maps/ devices/ nand/ onenand/ tests/
 
 obj-$(CONFIG_MTD_UBI)		+= ubi/
+obj-$(CONFIG_MTD)		+= ../../arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/
+
diff --git a/drivers/mtd/maps/Makefile b/drivers/mtd/maps/Makefile
index 6d4c957..252f334 100644
--- a/drivers/mtd/maps/Makefile
+++ b/drivers/mtd/maps/Makefile
@@ -58,7 +58,7 @@ obj-$(CONFIG_MTD_GPIO_ADDR)	+= gpio-addr-flash.o
 obj-$(CONFIG_MTD_LATCH_ADDR)	+= latch-addr-flash.o
 obj-$(CONFIG_MTD_LANTIQ)	+= lantiq-flash.o
 
-ifeq ($(CONFIG_ARCH_ARMADA_XP),y)
-	include $(srctree)/arch/arm/mach-armadaxp/config/mvRules.mk
-	obj-$(CONFIG_MV_INCLUDE_SPI) += ../../../arch/arm/mach-armadaxp/flashmap.o
+ifeq ($(CONFIG_MV_INCLUDE_SPI),y)
+	include $(srctree)/$(MACHINE)/config/mvRules.mk
+	obj-$(CONFIG_MV_INCLUDE_SPI) += ../../../$(MACHINE)/flashmap.o
 endif
diff --git a/drivers/mtd/nand/Makefile b/drivers/mtd/nand/Makefile
index 0bcd071..3402d59 100644
--- a/drivers/mtd/nand/Makefile
+++ b/drivers/mtd/nand/Makefile
@@ -52,6 +52,5 @@ obj-$(CONFIG_MTD_NAND_RICOH)		+= r852.o
 obj-$(CONFIG_MTD_NAND_JZ4740)		+= jz4740_nand.o
 obj-$(CONFIG_MTD_NAND_GPMI_NAND)	+= gpmi-nand/
 
-obj-$(CONFIG_MTD_NAND_NFC)			+= ../../../arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/
 
 nand-objs := nand_base.o nand_bbt.o
-- 
1.7.5.4

